streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -192,101 +194,117 @@ public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareActiveTask
      * the resulting assignment will have an absolute minimum of cross rack traffic. If we set {@code trafficCost} to 0,
      * and {@code nonOverlapCost} to a positive value, the resulting assignment will be identical to the input assignment.
      * <p>
+     * Note: this method will modify the input {@link KafkaStreamsAssignment} objects and return the same map.
+     * It does not make a copy of the map or the KafkaStreamsAssignment objects.
+     * <p>
      * This method optimizes cross-rack traffic for standby tasks only. For active task optimization,
      * use {@link #optimizeRackAwareActiveTasks}.
      *
      * @param kafkaStreamsAssignments the current assignment of tasks to KafkaStreams clients
      * @param applicationState        the metadata and other info describing the current application state
      *
-     * @return a new map containing the mappings from KafkaStreamsAssignments updated with the default rack-aware assignment for standy tasks
+     * @return a map with the KafkaStreamsAssignments updated to minimize cross-rack traffic for standby tasks
      */
-    public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTasks(
-        final ApplicationState applicationState,
-        final Map<ProcessId, KafkaStreamsAssignment> kafkaStreamsAssignments
-    ) {
+    private static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTasks(final ApplicationState applicationState,
```

Comment:
fyi I just made this private instead of fully removing it since it's partially implemented. This way we don't have to remove the whole thing if we don't get around to finishing the implementation. (And I filed a ticket so we don't forgot to change it back to public if/when we do implement it)